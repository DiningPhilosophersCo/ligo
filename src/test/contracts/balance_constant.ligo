(*
This test makes sure that the balance is accessible in PascaLIGO.

It is meant to detect the regression detailled in the following issue: https://gitlab.com/ligolang/ligo/issues/68
*)

type parameter is unit
type storage is tez
type return is list (operation) * storage

function main (const param : parameter; const store: storage) : return is
  ((nil : list (operation)), balance)
