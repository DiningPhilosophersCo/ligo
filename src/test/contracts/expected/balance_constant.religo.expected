type storage = tez;

let main2 = 
  (p: unit, s: storage) => 
    ([] : list(operation), Tezos.balance);

let main = (x: (unit, storage)) => main2(x[0], x[1]);
